### DISCUSSION TITLE
Using USR 304

### DISCUSSION THREAD

**Discussion by pgmelo1** [nathanmarlor/foxess_modbus#314] [General]
Anybody using the RS485 to Ethernet USR TCP232 - 304 does it work with H1 inverter modbus?

**Comment by canton7** [comment]
I haven't heard of anyone using that. Looking at the website quickly it looks like it should work though.

If you've got one, why not try it? If you don't have one, why not get one of the better alternatives?

**Reply by pgmelo1** [reply]
thank you for your response, I have one and it's connected but can't seem to get anything from it (modbus port 502 and inverter id is 247 - confirmed). Can access it's configuration and tried various settings but nothing. I can't open the plug on the inverter to see if it wasn't a issue connecting the wires. Any suggestions please?

**Reply by canton7** [reply]
What exact error message do you get (either in modpoll, or from the integration)?

What exact settings have you used? Can you post screenshots?

**Reply by pgmelo1** [reply]
Below screenshots and tried to with a command line tester (modpoll):
modpoll.exe -m tcp -p 502 -a 247 -t 3 -0 -1 -c 16 -r 11000 192.168.100.45
modpoll 3.10 - FieldTalk(tm) Modbus(R) Master Simulator
Copyright (c) 2002-2021 proconX Pty Ltd
Visit https://www.modbusdriver.com/ for Modbus libraries and tools.

Protocol configuration: MODBUS/TCP, FC4
Slave configuration...: address = 247, start reference = 11000 (PDU), count = 16
Communication.........: 192.168.100.45, port 502, t/o 1.00 s, poll rate 1000 ms
Data type.............: 16-bit register, input register table

-- Polling slave...
Reply time-out!

<img width="637" alt="Screenshot 2023-06-07 at 09 21 55" src="https://github.com/nathanmarlor/foxess_modbus/assets/102658156/2fb3a5f4-2d0b-4691-965c-9f4eaa1fe5f3">
<img width="830" alt="Screenshot 2023-06-07 at 09 20 07" src="https://github.com/nathanmarlor/foxess_modbus/assets/102658156/1500f385-fe7b-4fd0-a12e-300a596ea0f6">

**Reply by canton7** [reply]
Can you post screenshots of all of the configuration screens?

**Reply by pgmelo1** [reply]
<img width="806" alt="Screenshot 2023-06-07 at 09 31 27" src="https://github.com/nathanmarlor/foxess_modbus/assets/102658156/ff0b09e1-8525-4321-a9b5-68c9824b7276">
<img width="805" alt="Screenshot 2023-06-07 at 09 31 11" src="https://github.com/nathanmarlor/foxess_modbus/assets/102658156/3303f2d4-f609-42e6-8962-f0dfde046bfa">
<img width="805" alt="Screenshot 2023-06-07 at 09 30 56" src="https://github.com/nathanmarlor/foxess_modbus/assets/102658156/3293c260-91d6-4bd2-8944-2e749a887501">

**Reply by canton7** [reply]
Looking some more, I can't see any indication that this device supports modbus. With the other USR devices there are options on whether it's talking ModbusTCP <-> ModbusRTU etc, but I don't see anything like that with your device.

Try modpoll with `-m enc` rather than `-m tcp`?

**Reply by pgmelo1** [reply]
thank you! I'm assuming that the connections are ok but USR 304 doesn't work, I will order USR 410s.
the response with - m enc is:

modpoll.exe -m enc -p 502 -a 247 -t 3 -0 -1 -c 16 -r 11000 192.168.100.45
modpoll 3.10 - FieldTalk(tm) Modbus(R) Master Simulator
Copyright (c) 2002-2021 proconX Pty Ltd
Visit https://www.modbusdriver.com/ for Modbus libraries and tools.

Protocol configuration: Encapsulated RTU over TCP, FC4
Slave configuration...: address = 247, start reference = 11000 (PDU), count = 16
Communication.........: 192.168.100.45, port 502, t/o 1.00 s, poll rate 1000 ms
Data type.............: 16-bit register, input register table

-- Polling slave...
[11000]: 2916
[11001]: 14
[11002]: 410
[11003]: 882
[11004]: 0
[11005]: 2
[11006]: 2129
[11007]: 0
[11008]: -2
[11009]: 2327
[11010]: 17
[11011]: 360
[11012]: -29
[11013]: 401
[11014]: 4995
[11015]: 0

**Reply by canton7** [reply]
Ah OK, that looks promising.

The integration doesn't currently support the USR 304 in that case, but it might well be possible to make it work. If you're prepared to wait a day or two, I can create a beta for you to test?

**Reply by pgmelo1** [reply]
thank you, you made my day. After a yesterday the wind broke the structure of 4 panels ( the installer will take responsibility but was a very scary moment), I'll wait for your help.

**Reply by canton7** [reply]
Can you give v1.6.1b1 a try?

**Reply by pgmelo1** [reply]
It's working!!! I don't know how to thank you. If one day you come to Azores send na e-mail to at least take a beer with you :)
I use all auto to configure the energy dashboard, it's not showing all but I take it's normal. Check one entity and it show values :)
THANK YOU!!!

**Reply by canton7** [reply]
Awesome!

Would you mind filling in the "Setup" section of https://github.com/nathanmarlor/foxess_modbus/wiki/USR-TCP232-304 with instructions on how you configured the 304? See the [410s](https://github.com/nathanmarlor/foxess_modbus/wiki/USR-TCP232-410s) and [W610](https://github.com/nathanmarlor/foxess_modbus/wiki/USR-W610) for inspiration. That would really help other people with that device.

Thanks!

**Comment by pgmelo1** [comment]
Yes I would do it, can't make it right away but later today I would do it.
Just another think I used the auto function to the energy dashboard but to
see anything in the Solar productions I had to change the entity to Solar
Generation Today and delete PV1 and PV2 Power.

Thank you!
Pedro

**Reply by canton7** [reply]
Thanks!

I'm guessing the PV1/PV2 Energy Total sensors are still showing Unknown? Those are HA integration sensors, and it can take a while for them to start showing proper values, especially if you aren't getting PV power readings. See [the FAQs](https://github.com/nathanmarlor/foxess_modbus/wiki/FAQs#why-do-pv1-power-total--grid-consumption-total--feed-in-total--etc-show-unknown)

**Reply by dreampi** [reply]
I also ran into this issue, today. Maybe something to do with the update. Originally I was using pv1/pv2 total. But today I had to switch everything to solar generation today

**Reply by canton7** [reply]
What exactly is the issue? What do the PV1 / PV2 Power Total sensors show?

(they're fine for me, for what it's worth)

**Reply by dreampi** [reply]
The Values for them have not updated for the past 18 hours

**Reply by canton7** [reply]
So they have values? Can you open PV1 Power Total, expand the attributes section at the bottom, and screenshot?

**Reply by dreampi** [reply]
Yes, they have values. 
![046C0A96-8BF6-4922-9390-E3020D830DC3](https://github.com/nathanmarlor/foxess_modbus/assets/78707209/239c151e-ed40-443f-87eb-44fc6225e85f)

**Reply by canton7** [reply]
Interesting. You've got a friendly name there ("FoxESS") but the source sensor doesn't have a prefix. Can you click on your PV1 Power sensor and screenshot that?

**Reply by dreampi** [reply]
![DB59B6C2-4102-4776-BB42-50862C9EFD01](https://github.com/nathanmarlor/foxess_modbus/assets/78707209/6b22b5e0-de2b-44e3-bc0f-4dec7269bc84)

**Reply by canton7** [reply]
Oh sorry, click the cog (top right) first please?

**Reply by dreampi** [reply]
Here you go
![AB9C77B1-5DF3-4C28-A5AC-2D7179017FA5](https://github.com/nathanmarlor/foxess_modbus/assets/78707209/c9a351fe-a381-401a-a10e-fb0a068ea3b1)

**Reply by canton7** [reply]
Got it, thanks! Fix incoming...

**Reply by canton7** [reply]
Fixed in 1.6.1, thanks for reporting and helping me narrow it down.

**Reply by dreampi** [reply]
Your welcome. You’ve probably noticed already but I’m getting the same issue with the Load total

**Reply by canton7** [reply]
Yep, and all of the other Riemann sum integral sensors
